Tests/ToDoListTests/ToDoListTests.swift

```
@@ -4,64 +4,83 @@ import XCTest
+        self.file = File("Filename", atPath: path, withTodos: [todo])
+    }
+    
     //Read inputs
-    func testInputsAreReadable() throws {
+    func testInputIsReadable() throws {
         let reader = Reader()
         let input = Input()
         let result = input.isReadableBy(reader)
         XCTAssertTrue(result)
     }
     
-    //Show outputs
-    func testOutPutsAreShown() throws {
-        XCTFail()
-    }
+//    //Show outputs
+//    func testOutPutsAreShown() throws {
+//        XCTFail()
+//    }
     
     //Create the File
     func testCreateAFileWithSpecificPath() throws {
-        let path = ""
-        let file = File("fileName", atPath: path)
         let result = try file.create()
         XCTAssertTrue(result)
     }
     
     //Write in the file
     func testWriteInAFile() throws {
         let jsonContent = JSONContent()
-        let path = ""
-        let file = File("filename", atPath: path)
         let writer = Writer()
         let result = writer.write(jsonContent, in: file)
         XCTAssertTrue(result)
     }
     
     //Delete the file
-    func testDeleteAfile() throws {
-        XCTFail()
+    func testDeleteFileAtPath() throws {
+        let result = try file.delete()
+        XCTAssertTrue(result)
     }
     
     //Add a todo
-    func testAddATodo() throws {
-        XCTFail()
+    func testAddATodoToFile() throws {
+        let result = try file.add(self.todo)
+        XCTAssertTrue(result)
     }
     
     //Update a todo
-    func testUpdateATodo() throws {
-        XCTFail()
+    func testUpdateATodoFromFile() throws {
+        let todos = self.file.getTodos()
+        guard var todo = todos.first else { return XCTFail() }
+        todo.setSubject("toto")
+        let result = try file.update(todo)
+        XCTAssertTrue(result)
     }
     
-    //Delete a todo
-    func testDeleteATodo() throws {
-        XCTFail()
+//    //Delete a todo
+    func testDeleteATodoFromFile() throws {
```

Do your test really need to throw an exception?